api/common/interfaces.go

```diff
@@ -491,6 +491,40 @@ func getNonceByAddr(s Serverer, params map[string]interface{}) map[string]interf
 	return respPacking(ret, SUCCESS)
 }
 
+// getId gets id by publick key
+// params: {"publickey":<publickey>}
+// return: {"result":<result>, "error":<errcode>}
+func getId(s Serverer, params map[string]interface{}) map[string]interface{} {
+	if len(params) < 1 {
+		return respPacking(nil, INVALID_PARAMS)
+	}
+
+	publicKey, ok := params["publickey"].(string)
+	if !ok {
+		return respPacking(nil, INVALID_PARAMS)
+	}
+
+	pkSlice, err := hex.DecodeString(publicKey)
+	if err != nil {
+		return respPacking(nil, INVALID_PARAMS)
+	}
+
+	id, err := chain.DefaultLedger.Store.GetID(pkSlice)
+	if err != nil {
+		return respPacking(nil, INVALID_PARAMS)
+	}
+
+	if len(id) == 0 || bytes.Equal(id, make([]byte, 32)) {
+		return respPacking(nil, INTERNAL_ERROR)
```

Comment:
We need to have a special error code if pk does not have an id yet so that the node knows he needs to generate an ID